added mongoose hooks for removing email case sensitivity

mannan-b opened the pull request.

mannan-b committed:
added mongoose hooks for removing email case sensitivity

Shobhit141141 commented:
Handle email case sensitivity at login as well (at controller level), keep original casing untouched in past MongoDB records.

Remove the migration code

Shobhit141141 commented on src/api/models/migrateEmail.ts:
Remove this

mannan-b committed:
Fix(email): Handle email case sensitivity at controller level.

- Removed email migration script.
- Reverted User model hooks (email is now stored as-is).
- Updated UserService to use case-insensitive regex for lookups.
- Updated .gitmodules to use HTTPS.

godaralokesh29 commented:
1.Fix migration import/export
Problem.. migrateEmail.ts imports "./User" but the model is exported from userModel.ts as a named export.
change to.. import { User } from "./userModel";

2.Make migration safe 
Problem: current script calls process.exit() immediately and lacks a dry-run.....Don’t exit without disconnecting...
Await mongoose.disconnect() (in finally) and use proper exit codes.
The script calls process.exit() immediately; better to close the mongoose connection first:
await mongoose.disconnect(); then process.exit(0).
This avoids potential unflushed logs or hanging connections.

mannan-b commented:
the migrate email thing was completely removed in the last commit, so i guess it wont be an issue now

godaralokesh29 commented:
in src/api/services/userService.ts ok to keep case-insensitive detection... ensure we do not mutate the email when creating users. Move escapeRegex to a shared util.Do NOT change the stored email casing...Lookups should be case insensitive at controller level..do not lowercase in model

mannan-b committed:
Fix(email): Handle email case sensitivity at controller level

godaralokesh29 commented:
good good .....just small tweaks needed ..between findExistingUsersByEmail() and insertMany(), another process could insert the same email insertMany throws a bulk write error...Wrap insertMany in try/catch and use ordered: false so one duplicate doesn’t block other inserts
....Please also test this locally. Does this meet the intended purpose when run locally???????

mannan-b committed:
Fix(email): Enforce case-insensitive uniqueness in controllers and scripts. Address PR #137

mannan-b committed:
Fix: Revert .gitmodules URL and format code

mannan-b committed:
Refactor: Enforce lowercase email via pre-save hook, remove regex

mannan-b committed:
Removed test scripts and changes in userService

mannan-b committed:
Delete scripts/verify-email-auth.ts

mannan-b committed:
Cleanup of controllers and services

mannan-b committed:
resolved merge conflicts and cleanup

mannan-b committed:
Delete scripts/verify-email-auth.ts

mannan-b committed:
Cleanup of userController

godaralokesh29 commented on src/api/controllers/userController.ts:
```diff
@@ -181,14 +181,50 @@ export const postBulkUserController = asyncErrorHandler(
+      } else {
+        throw error;
+      }
+    }
```
remove these  line

godaralokesh29 commented on src/api/controllers/userController.ts:
```diff
@@ -181,14 +181,50 @@ export const postBulkUserController = asyncErrorHandler(
       })),
     );
 
-    const userDocsInserted = await User.insertMany(hashedUsers);
```
keep this .. dont remove this

godaralokesh29 commented on src/api/controllers/userController.ts:
```diff
@@ -254,7 +290,22 @@ export const updateUserController = asyncErrorHandler(
+        return next(error);
+      }
+      user.email = req.body.target_email;
+    }
```
remove these

godaralokesh29 commented on src/api/controllers/userController.ts:
```diff
@@ -254,7 +290,22 @@ export const updateUserController = asyncErrorHandler(
 
     // Update user properties if provided in request body
     if (req.body.target_name) user.name = req.body.target_name;
-    if (req.body.target_email) user.email = req.body.target_email;
```
keep this

VyomRaturi committed:
linting errors fixed
